client/src/ui/organisms/header/index.tsx

```diff
@@ -51,8 +51,22 @@ export function Header() {
     return false;
   }
 
+  const [isProgressBarVisible, setProgressBarVisible] = useState(false);
+  useEffect(
+    () =>
+      window.addEventListener("beforeunload", () =>
+        setProgressBarVisible(true)
+      ),
```

Comment:
Please correct me if I'm wrong, but to avoid adding a new event listener (when re-rendering the component) without removing up the previous one, you might want to return a clean up callback:

```suggestion
      function handleBeforeUnload() {
        setProgressBarVisible(true)
      }

      window.addEventListener("beforeunload", handleBeforeUnload)
      
      return () => window.removeEventListener("beforeunload", handleBeforeUnload)
```